zds/search2/models.py

```diff
@@ -0,0 +1,283 @@
+# coding: utf-8
+from django.db import models
+from django.apps import apps
+from django.db.transaction import atomic
+
+from elasticsearch.helpers import streaming_bulk
+from elasticsearch_dsl import Mapping
+from elasticsearch_dsl.connections import connections
+
+
+class ESIndexableMixin(object):
+    """Mixin for indexable objects.
+
+    Define a number of different functions that can be overridden to tune the behavior of indexing into elasticsearch.
+
+    You (may) need to override :
+
+    - ``get_indexable()`` ;
+    - ``get_mapping()`` (not mandatory, but otherwise, ES will choose the mapping by itself) ;
+    - ``get_document()`` (not mandatory, but may be useful if data differ from mapping or extra stuffs need to be done).
+
+    You also need to maintain ``es_id`` and ``es_already_indexed`` for bulk indexing/updating (if any).
+    """
+
+    es_already_indexed = False
+    es_id = ''
+
+    @classmethod
+    def get_es_content_type(cls):
+        """value of the ``_type`` field in the index"""
+        content_type = cls.__name__.lower()
+
+        # fetch parents
+        for base in cls.__bases__:
```

Comment:
serait-il possible d'ajouter un `logger.debug("will iter into %s", cls.__bases__)` avant la boucle et un `logger.debug("content type is %", content_type")` lorsque la boucle est matchée.

de plus en mode "micro optimisation mais au moins on comprend explicitement la volonté du développeur", peux-tu ajouter un "break" s'il te plait?